resources/src/commonMain/kotlin/ThemedColorProvider.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+@file:JvmName("CommonThemedColorProvider")
+package com.splendo.kaluga.resources
+
+import kotlin.jvm.JvmName
+import kotlin.reflect.KProperty
+
+/** Multiplatform Color Provider that is aware of a Dark and Light mode */
+abstract class ThemedColorProvider {
+
+    /** light version of this color */
+    abstract val lightColor: Color
+
+    /** dark version of this color */
+    abstract val darkColor: Color
+
+    /** main color, automatically changing based on theme */
+    abstract val color: Color
+}
+
+/** This extension allows to get the hexValue of the provided color with the "by" syntax */
+inline operator fun ThemedColorProvider.getValue(thisRef: Any?, property: KProperty<*>): Color = color
+
+/**
+ * One single color represented in 2 themes Dark or Light.
+ * @param light color when presented in Light mode: formatted as hexadecimal string
+ * @param dark color when presented in Dark mode: formatted as hexadecimal string
+ */
+expect fun themeColor(light: String, dark: String): ThemedColorProvider
```

Comment:
There is an isInDarkMode method on the kaluga level. So I dont see the point of making this platform specific. It should just be:

fun themeColor(light: Color, dark: Color): Color {
   if (isInDarkMode) dark else light
}

Reply:
Also, if you want the dynamic behaviour on Android on a named level, you can just make a values_night folder and put the dark color there.

Reply:
The main reason that simple if statement doesn't reflect change in real time on iOS (including SwiftUI Previews) so UIColor should be created on other way using special constructor to have dynamic colors.

Reply:
No, look at the implementation. iOS provides a color that is both light and dark, whereas this does not exist on Android. See https://splendo.slack.com/archives/CN1T4SM5Y/p1618294377023400

Reply:
Actually good point regarding `values-night` worth to try on android and just returning same color name to be handled by Android itself?

Reply:
putting it in `values-night` defies the point of what this method tries to do, defining colors once in common code.

Reply:
> Actually good point regarding `values-night` worth to try on Android and just returning same color name to be handled by Android itself?

Yes, this is worth looking into and too bad I didn't think about it in the first place.
I was doubting that values-night is the folder used in case of dark mode (earlier versions on Android would use this folder for other modes), but indeed it is.

I will make some more experiments when I have time

@tijl I agree that the preferred usage of this class is indeed by using the strings directly.
Although the problem is trying to solve is also to unify the usage of colors on platform without breaking the dark/light switch.
In that sense, I think is worth looking into the values-night solution while still using these ColorProviders, we would only need to add 1 extension to accepts colors directly. It will still be valuable the fact that we define the difference in the constructor.
I think this is useful for project where the best/only solution to branded colors is with scripts that generates the XML files in the correct places.
In general, we should consider how to support projects that already have their color definitions in xml and don't need to change that,  while they do need to add support fro dark/light easily
(imho)

Reply:
Can you stop tagging me please